javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/resolution/typesolvers/ClassLoaderTypeSolver.java

```diff
@@ -39,9 +41,60 @@ public class ClassLoaderTypeSolver implements TypeSolver {
 
     private TypeSolver parent;
     private ClassLoader classLoader;
+    private HashMap<String, Set<String>> modulePackages;
 
     public ClassLoaderTypeSolver(ClassLoader classLoader) {
         this.classLoader = classLoader;
+        this.modulePackages = new HashMap<>();
+    }
+
+    /**
+     * Create a ClassLoaderTypeSolver with a list of module layers to check when solving types in modules. If
+     * moduleLayers is empty, tryToSolveTypeInModule will always return SymbolReference.unsolved
+     *
+     * @param classLoader the ClassLoader that should be used for type solving
+     * @param moduleLayers MUST be {@code Iterable<java.lang.ModuleLayer>}. Object is only used in the signature for
+     *                     Java 8 compatibility.
+     */
+    public ClassLoaderTypeSolver(ClassLoader classLoader, Iterable<Object> moduleLayers) {
+        this(classLoader);
+        addModuleLayers(moduleLayers);
+    }
+
+    public void addModuleLayers(Iterable<Object> moduleLayers) {
+        for (Object moduleLayer : moduleLayers) {
+            if (!moduleLayer.getClass().getCanonicalName().equals("java.lang.ModuleLayer")) {
+                continue;
+            }
+            try {
+                /* This code is equivalent to the snippet below, but is done with reflection to maintain compatibility
```

Comment:
Can you isolate this code in a class that could be called, for example, ModuleLayerHelper?